### ISSUE TITLE
Error due to issues in the SCIP model.

### ISSUE THREAD

**Issue by kartikeyrinwa** [ZIB-IOL/Boscia.jl#171]
When I run the the Boscia package with a MOI oracle I am getting an error from SCIP. 
I get this error in two locations. Once in the check_feasibility() method in the MOI_bounded_oracle.jl file and  once in the compute_extreme_point method. 
I have an example of the error below. 
I was able to fix the issues by adding a call to the function free_model(blmo.o) before the SCIP changing functions are called. However, this required changes to the Boscia files with themselves.  

```
[scip_var.c:4551] ERROR: invalid SCIP stage <9>
ERROR: LoadError: SCIPchgVarObj(o, var(o, change.variable), change.new_coefficient) yielded SCIP code SCIP_INVALIDCALL
Stacktrace:
  [1] error(s::String)
    @ Base ./error.jl:33
  [2] macro expansion
    @ ~/.julia/packages/SCIP/S9mBb/src/wrapper.jl:14 [inlined]
  [3] modify(o::SCIP.Optimizer, #unused#::MathOptInterface.ObjectiveFunction{MathOptInterface.ScalarAffineFunction{Float64}}, change::MathOptInterface.ScalarCoefficientChange{Float64})
    @ SCIP ~/.julia/packages/SCIP/S9mBb/src/MOI_wrapper/objective.jl:70
  [4] compute_extreme_point(lmo::FrankWolfe.MathOptLMO{SCIP.Optimizer}, direction::Vector{Float64}; kwargs::Base.Iterators.Pairs{Symbol, Base.Iterators.Pairs{Symbol, Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}}, Tuple{Symbol}, NamedTuple{(:kwargs,), Tuple{Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}}}}}, Tuple{Symbol}, NamedTuple{(:kwargs,), Tuple{Base.Iterators.Pairs{Symbol, Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}}, Tuple{Symbol}, NamedTuple{(:kwargs,), Tuple{Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}}}}}}}})
    @ FrankWolfe ~/.julia/packages/FrankWolfe/O0VKB/src/moi_oracle.jl:32
  [5] #compute_extreme_point#53
    @ ~/.julia/dev/Boscia/src/MOI_bounded_oracle.jl:39 [inlined]
  [6] compute_extreme_point(tlmo::Boscia.TimeTrackingLMO{Boscia.MathOptBLMO{SCIP.Optimizer}}, d::Vector{Float64}; kwargs::Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ Boscia ~/.julia/dev/Boscia/src/time_tracking_lmo.jl:34
  [7] compute_extreme_point
    @ ~/.julia/dev/Boscia/src/time_tracking_lmo.jl:32 [inlined]
  [8] blended_pairwise_conditional_gradient(f::NDFW.var"#f#43"{String, NDFW.TA_Data, Vector{Float64}, Float64, Int64, Int64, Int64, Int64}, grad!::NDFW.var"#grad!#44"{String, NDFW.TA_Data, Vector{Float64}, Float64, Int64, Int64, Int64, Int64}, lmo::Boscia.TimeTrackingLMO{Boscia.MathOptBLMO{SCIP.Optimizer}}, active_set::FrankWolfe.ActiveSet{Vector{Float64}, Float64, Vector{Float64}}; line_search::FrankWolfe.Adaptive{Float64, Int64}, epsilon::Float64, max_iteration::Int64, print_iter::Int64, trajectory::Bool, verbose::Bool, memory_mode::FrankWolfe.InplaceEmphasis, gradient::Nothing, callback::Boscia.var"#fw_callback#19"{Bonobo.BnBTree{Boscia.FrankWolfeNode{FrankWolfe.ActiveSet{Vector{Float64}, Float64, Vector{Float64}}, FrankWolfe.DeletedVertexStorage{Vector{Float64}}, Boscia.IntegerBounds}, NamedTuple{(:problem, :current_node_id, :updated_incumbent, :global_tightening_rhs, :global_tightening_root_info, :global_tightenings, :options), Tuple{Boscia.SimpleOptimizationProblem{NDFW.var"#f#43"{String, NDFW.TA_Data, Vector{Float64}, Float64, Int64, Int64, Int64, Int64}, NDFW.var"#grad!#44"{String, NDFW.TA_Data, Vector{Float64}, Float64, Int64, Int64, Int64, Int64}, Boscia.TimeTrackingLMO{Boscia.MathOptBLMO{SCIP.Optimizer}}, Boscia.IntegerBounds}, Base.RefValue{Int64}, Base.RefValue{Bool}, Base.RefValue{Float64}, NamedTuple{(:lower_bounds, :upper_bounds), Tuple{Dict{Int64, Tuple{Float64, Float64}}, Dict{Int64, Tuple{Float64, Float64}}}}, Boscia.IntegerBounds, Dict{Symbol, Any}}}, Vector{Float64}, Boscia.FrankWolfeSolution{Boscia.FrankWolfeNode{FrankWolfe.ActiveSet{Vector{Float64}, Float64, Vector{Float64}}, FrankWolfe.DeletedVertexStorage{Vector{Float64}}, Boscia.IntegerBounds}, Vector{Float64}}}, Float64, Bool, Vector{Int64}, Int64, Vector{MathOptInterface.VariableIndex}}, traj_data::Vector{Any}, timeout::Int64, renorm_interval::Int64, lazy::Bool, linesearch_workspace::Nothing, lazy_tolerance::Float64, extra_vertex_storage::FrankWolfe.DeletedVertexStorage{Vector{Float64}}, add_dropped_vertices::Bool, use_extra_vertex_storage::Bool, recompute_last_vertex::Bool)
    @ FrankWolfe ~/.julia/packages/FrankWolfe/O0VKB/src/pairwise.jl:277
```

**Comment by dhendryc** [comment]
What do you mean by "SCIP changing functions"? The change in the linear objective function in every FW iteration?

**Comment by dhendryc** [comment]
Can you try your example with the `free-scip-model` branch?

**Comment by dhendryc** [comment]
Should be fixed in #170
